File: solutions/_905_sort_array_by_parity/solution.py
from typing import List


class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        L = 0
        for R in range(len(nums)):
            if nums[R] % 2 == 0:
                nums[L], nums[R] = nums[R], nums[L]
                L += 1
        return nums


# https://leetcode.com/problems/sort-array-by-parity/

File: solutions/_905_sort_array_by_parity/solution.cpp
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> sortArrayByParity(vector<int> &nums)
    {
        int L = 0;
        for (int R = 0; R < nums.size(); R++)
        {
            if (nums[R] % 2 == 0)
            {
                swap(nums[L++], nums[R]);
            }
        }
        return nums;
    }
};

/*
https://leetcode.com/problems/sort-array-by-parity/
*/